### PR TITLE
Complement color

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### Reference Issues/PRs
#122
#### What does this implement/fix? Explain your changes.
- `is_valid_color` function added
- `color_complement` function added
- `select_color` function added
- Complementary color support for `color` and `bgcolor` parameters
- `filter_color` function modified

### CODE REVIEW

File: samila/functions.py [hunk 1]
```diff
@@ -95,17 +95,70 @@ def distance_calc(s1, s2):
...
+
+def color_complement(color):
+    """
+    Calculate complement color.
+
+    :param color: given color
+    :type color: any format
+    :return: complement color
+    """
+    color = color[1:]
```
Comment (sadrasabouri) [inline samila/functions.py:121]: Why is that for?

File: samila/functions.py [hunk 2]
```diff
@@ -95,17 +95,70 @@ def distance_calc(s1, s2):
...
+
+def filter_color(color, bgcolor):
     """
     Filter given color and return it.
 
     :param color: given color
-    :type color: str or tuple
-    :return: filtered version of color
+    :type color: any format
+    :param bgcolor: giver background color
```
Comment (sadrasabouri) [inline samila/functions.py:134]: `given`

File: test/function_test.py [hunk 3]
```diff
@@ -13,16 +13,16 @@
...
+>>> filter_color((0.2,0.3,0.4),(0.2,0.3,0.4,1))
+((0.2, 0.3, 0.4), (0.2, 0.3, 0.4, 1))
+>>> filter_color("#FFFFFF","#ffffe1")
+('#FFFFFF', '#ffffe1')
 >>> random.seed(2)
->>> color1 = filter_color("random")
+>>> color1, bgcolor1 = filter_color("random","random")
 >>> random.seed(3)
->>> color2 = filter_color("RANDOM")
+>>> color2,bgcolor2 = filter_color("RANDOM","RANDOM")
```
Comment (sadrasabouri) [inline test/function_test.py:25]: `color2, bgcolor2`

Comment (sepandhaghighi) [reply test/function_test.py]: fixed in https://github.com/sepandhaghighi/samila/pull/131/commits/8371989046bc9748c389876c91d8a2049fdce471

File: samila/functions.py (hunk 2 above)
Comment (sepandhaghighi) [reply samila/functions.py]: fixed in https://github.com/sepandhaghighi/samila/pull/131/commits/17d8cf95f14f968890c1ce71d24b1c755b3a99cb

File: samila/functions.py (hunk 1 above)
Comment (sepandhaghighi) [reply samila/functions.py:121]: To remove the `#` symbol.
```pycon
>>> color = "#ffffff"
>>> color[1:]
'ffffff'
```

File: test/function_test.py [hunk 4]
```diff
@@ -13,16 +13,16 @@
...
+>>> filter_color((0.2,0.3,0.4),(0.2,0.3,0.4,1))
+((0.2, 0.3, 0.4), (0.2, 0.3, 0.4, 1))
+>>> filter_color("#FFFFFF","#ffffe1")
+('#FFFFFF', '#ffffe1')
 >>> random.seed(2)
->>> color1 = filter_color("random")
+>>> color1, bgcolor1 = filter_color("random","random")
 >>> random.seed(3)
->>> color2 = filter_color("RANDOM")
+>>> color2, bgcolor2 = filter_color("RANDOM","RANDOM")
```
Comment (sadrasabouri) [inline test/function_test.py:25]: `filter_color("RANDOM", "RANDOM")`

File: test/function_test.py [hunk 5]
```diff
@@ -13,16 +13,16 @@
...
 False
->>> filter_color("yellow")
-'yellow'
->>> filter_color((0.2,0.3,0.4))
-(0.2, 0.3, 0.4)
->>> filter_color("#FFFFFF")
-'#FFFFFF'
+>>> filter_color("yellow", "blue")
+('yellow', 'blue')
+>>> filter_color((0.2,0.3,0.4),(0.2,0.3,0.4,1))
```
Comment (sadrasabouri) [inline test/function_test.py:18]: `filter_color((0.2,0.3,0.4), (0.2,0.3,0.4,1))`

**Review by sadrasabouri — APPROVED** [review]
It seems OK to me.

**Review by sadrasabouri — APPROVED** [review]

**Pull request merged**